MIDAS/src/hardware/GPSSensor.cpp

```diff
@@ -19,12 +19,12 @@ ErrorCode GPSSensor::init() {
 
 // This is needed because GPS doesn't provide unix time and just gives
 // dd mm yy
-uint16_t months[12] = {
+const uint16_t months[12] = {
     0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334
 };
 
 inline bool is_leapyear(int year) {
-    return (!(year % 100 == 0) || (year % 400 == 0)) && year % 4 == 0;
+    return (!(year % 100) || (year % 400 == 0)) && year % 4 == 0;
```

Comment:
i think it'd would have to be `(year % 100)`, not `!(year % 100)`. But please just use `(year % 100 != 0)`.